app/controllers/movement/MovementSummaryController.scala

```diff
@@ -63,45 +56,34 @@ class MovementSummaryController @Inject()(
     }
   }
 
+  private def parseDUCR(ucrBlock: UcrBlock): Option[String] =
+    ucrBlock.ucrType match {
+      case "D" => Some(ucrBlock.ucr)
+      case _   => None
+    }
+
   def submitMovementRequest(): Action[AnyContent] = authenticate.async { implicit request =>
     customsCacheService.fetchMovementRequest(movementCacheId, request.user.eori).flatMap {
       case Some(data) => {
+        val ducrVal = parseDUCR(data.ucrBlock).getOrElse("")
         val eoriVal = request.user.eori
```

Comment:
Just to remember for the future to change naming here, because it's not related with your changes. `ducrVal` -> `ducr`

I didn't see reason of this Val

Reply:
This whole section of code needs refactoring it was put there so its plainly visible what values are being sent to the backend. Mainly this was to get it working as was. I'll add a TODO on here so its very clear